{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "hindu-parliament",
   "metadata": {},
   "source": [
    "# Cox proportional hazards models for T-ALL paper\n",
    "\n",
    "Select kernel: Single Cell R\n",
    "\n",
    "* Fig4D & SuppTable 11\n",
    "* Fig4E & SuppTable 13\n",
    "* SuppFig6A & SuppTable12"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "shaped-verse",
   "metadata": {},
   "source": [
    "### Set up"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "supported-metro",
   "metadata": {},
   "outputs": [],
   "source": [
    "### suppressMessages(source(\"/nfs/users/nfs_h/hw12/T_ALL/notebooks/T_helper_functions.R\"))\n",
    "requiredPackages <- c(\"tidyverse\", \"ggpubr\",\"patchwork\",\"ggrepel\",\n",
    "                     \"circlize\",\"ComplexHeatmap\", \"GSEABase\", \"edgeR\",\"limma\",\n",
    "                     \"survminer\",\"survival\", \"plyr\",\"ggalt\",\"ggbeeswarm\") #\"Manu\",\n",
    "\n",
    "for (pkg in requiredPackages){\n",
    "  suppressWarnings(suppressMessages(library(pkg, character.only = T)))\n",
    "}\n",
    "\n",
    "source(\"/nfs/users/nfs_h/hw12/processing_scripts/survival_analysis/GDC_signature_survival_functions.R\")\n",
    "source(\"/nfs/users/nfs_h/hw12/processing_scripts/survival_analysis/GDC_signature_survival_functions_final.R\")\n",
    "\n",
    "fig <- function(width, heigth){\n",
    " ## From - https://stackoverflow.com/questions/45473128/r-changing-ggplot-plot-size-in-jupyter\n",
    " options(repr.plot.width = width, repr.plot.height = heigth)\n",
    " }\n",
    "\n",
    "### --- Load modules\n",
    "Z_modules <- getGmt(paste0(\"/lustre/scratch127/recovered/scratch126/lustre/scratch126/casm/team274sb/hw12/T_ALL/modules/\", \n",
    "                     \"ZBTB16_modules_refined.gmt\"))\n",
    "### --- Load data\n",
    "#load(file=\"/lustre/scratch126/casm/team274sb/project_folders/T_ALL_paper/data/UPenn_dge.Rdata\")\n",
    "load( file=\"/nfs/team274/bl10/TALL_bulkRNA_UPenn/UPenn_dge.Rdata\")\n",
    "\n",
    "\n",
    "### --- Add survival info\n",
    "#load_dir <- \"/lustre/scratch126/casm/team274sb/bl10/T-ALL/Data/Bulk_RNA/UPennFINAL/\"\n",
    "load_dir = \"/nfs/team274/bl10/TALL_bulkRNA_UPenn/\"\n",
    "bulk_metadat_1 = read.table(paste0(load_dir,\"ST3_Genomic_Data.csv\"),sep=\",\", header=TRUE)\n",
    "bulk_metadat_2 = read.table(paste0(load_dir,\"ST1_Clinical_Data.csv\"),sep=\",\", header=TRUE)\n",
    "\n",
    "bulk_metadat_2$DueToDisease <- ifelse(bulk_metadat_2$Cause.of.Death %in% c(\"; Due to this disease\",\"Due to this disease;\",\n",
    "                                                                           \"Due to this disease\",\" Due to this disease\"),\n",
    "                                      \"Due to this disease\",\"Not\")\n",
    "\n",
    "\n",
    "\n",
    "for (col_str in c('OS','OS.status','EFS','EFS.status','DFS','DFS.status','DueToDisease')){\n",
    "    UPenn_dge$samples[[col_str]] <- as.vector(setNames(bulk_metadat_2[[col_str]],\n",
    "                                    bulk_metadat_2$USI)[UPenn_dge$samples$sample_id])\n",
    "}\n",
    "\n",
    "save_dir <- \"/lustre/scratch125/cellgen/behjati/hw12/T_ALL/survival_output/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "reflected-employer",
   "metadata": {},
   "outputs": [],
   "source": [
    "plotGDCSurvival_hjw <- function(survdf_list,n_row=1, str=\"ModuleScore\", y_str=\"\"){\n",
    "\n",
    "  plotdf = ldply(survdf_list, .id = 'Project')\n",
    "  \n",
    "  #order by pvals\n",
    "  pvaldf = base::unique(plotdf[, c('Project', 'p')])\n",
    "  plotdf$Project = factor(plotdf$Project, pvaldf$Project[order(pvaldf$p)])\n",
    "  \n",
    "  #add counts\n",
    "  plotdf = ddply(plotdf, 'Group', function (x) {\n",
    "    x$Group = paste0(x$Group, ' ~ ', str,' (n=', nrow(x), ')')\n",
    "    return(x)\n",
    "  })\n",
    "  \n",
    "  #pvalue annotations\n",
    "  annotdf = base::unique(plotdf[, c('Project', 'p_txt')])\n",
    "  annotdf$Group = plotdf$Group[1]\n",
    "  \n",
    "  p1 = ggplot(plotdf, aes(Time, surv, colour = Group, fill = Group)) +\n",
    "    geom_ribbon(\n",
    "      aes(ymin = lower, ymax = upper),\n",
    "      colour = NA,\n",
    "      alpha = 0.25,\n",
    "      stat = 'stepribbon',\n",
    "      show.legend = FALSE\n",
    "    ) +\n",
    "    geom_step(show.legend = FALSE) +\n",
    "    geom_point(data = plotdf[plotdf$Event != 0, ],\n",
    "               shape = '+',\n",
    "               size = 5) +\n",
    "    geom_text(\n",
    "      aes(x = -Inf, y = -Inf, label = p_txt),\n",
    "      hjust = -0.1,\n",
    "      vjust = -1,\n",
    "      colour = 'black',\n",
    "      data = annotdf,\n",
    "      show.legend = FALSE\n",
    "    ) +\n",
    "    facet_wrap(~ Project, scales = 'free_x',nrow=n_row) +\n",
    "    xlab('Time') +\n",
    "    scale_y_continuous(limits = c(0, 1),\n",
    "                       name = paste0('Survival probability',y_str),\n",
    "                       breaks = c(0, 0.5, 1)) +\n",
    "    scale_colour_manual(\n",
    "     # palette ='Set2',#'Set2',\n",
    "      values=c( \"#8E1F43\",\"#237186\", \"#DCC949\"),\n",
    "      aesthetics = c('colour', 'fill'),\n",
    "      # label = function(breaks) {parse(text = breaks)},\n",
    "      guide = guide_legend(\n",
    "        override.aes = list(shape = '+', linetype = 1),\n",
    "        ncol = 2,\n",
    "        title = ''\n",
    "      )\n",
    "    ) +\n",
    "    guides(fill = guide_none()) + theme_bw() +\n",
    "    theme(legend.position = 'bottom', axis.text=element_text(size=12),axis.title=element_text(size=16),\n",
    "          legend.text=element_text(size=14), strip.background = element_blank(),strip.text.x=element_text(size=18))\n",
    "  return(p1)\n",
    "}\n",
    "\n",
    "runSurvivalAnalysis <- function(cdata_x,cluster, event_str){\n",
    "    #cdata_x = UPenn_dge$samples\n",
    "    \n",
    "    ## -- Check values\n",
    "    if (length(cdata_x[which((cdata_x[, event_str] < 0) == TRUE), \n",
    "                   \"short_name\"]) > 0 ){\n",
    "        print(\"Some patients have negative values for event\")\n",
    "    }\n",
    "    \n",
    "    ## -- Create into survival data structure\n",
    "    cdata_x$s = grepl(\"1\", cdata_x$OS.status, ignore.case = TRUE)\n",
    "    cdata_x$cluster = as.factor(cluster)\n",
    "    cdata_x$event <- as.vector(cdata_x[[event_str]])\n",
    "    cdata_x = cdata_x[, c(\"event\", \"s\", \"cluster\")]\n",
    "    \n",
    "    ## -- Set formula\n",
    "    f = as.formula(\"Surv(event, s) ~ cluster\")\n",
    "    #f = as.formula(\"Surv(event) ~ cluster\")\n",
    "    sfit = do.call(survival::survfit, list(formula = f, data = cdata_x))\n",
    "    \n",
    "    ## -- Build DF\n",
    "    strata = sfit$strata\n",
    "    names(strata) = gsub('cluster=', '', names(strata))\n",
    "\n",
    "    strata = rep(names(strata), times = strata)\n",
    "    plotdf = data.frame(\n",
    "        'Time' = sfit$time,\n",
    "        'surv' = sfit$surv,\n",
    "        'Event' = sfit$n.censor,\n",
    "        'lower' = sfit$lower,\n",
    "        'upper' = sfit$upper,\n",
    "        'Group' = strata,\n",
    "        'p' = surv_pvalue(sfit, data = cdata_x)$pval,\n",
    "        'p_txt' = surv_pvalue(sfit, data = cdata_x)$pval.txt\n",
    "    )\n",
    "    #plotdf = plotdf[!is.na(plotdf$lower), ]\n",
    "    #stopifnot(all(table(plotdf$Group) > 1))\n",
    "    \n",
    "    survdfs = list(T_ALL=plotdf)\n",
    "    survdfs = survdfs[!is.na(survdfs)]\n",
    "    \n",
    "    ## -- Return\n",
    "    return(survdfs)\n",
    "    \n",
    "}\n",
    "\n",
    "runSurvivalAnalysis_EFS <- function(cdata_x, cluster, event_str, status_str){\n",
    "   # cdata_x = UPenn_dge$samples\n",
    "    \n",
    "    ## -- Check values\n",
    "    if (length(cdata_x[which((cdata_x[, event_str] < 0) == TRUE), \n",
    "                   \"short_name\"]) > 0 ){\n",
    "        print(\"Some patients have negative values for event\")\n",
    "    }\n",
    "    \n",
    "    ## -- Create into survival data structure\n",
    "    cdata_x$s = grepl(\"1\", cdata_x[[status_str]], ignore.case = TRUE)\n",
    "    cdata_x$cluster = as.factor(cluster)\n",
    "    cdata_x$event <- as.vector(cdata_x[[event_str]])\n",
    "    cdata_x = cdata_x[, c(\"event\", \"s\", \"cluster\")]\n",
    "    \n",
    "    ## -- Set formula\n",
    "    f = as.formula(\"Surv(event, s) ~ cluster\")\n",
    "    #f = as.formula(\"Surv(event) ~ cluster\")\n",
    "    sfit = do.call(survival::survfit, list(formula = f, data = cdata_x))\n",
    "    \n",
    "    ## -- Build DF\n",
    "    strata = sfit$strata\n",
    "    names(strata) = gsub('cluster=', '', names(strata))\n",
    "\n",
    "    strata = rep(names(strata), times = strata)\n",
    "    plotdf = data.frame(\n",
    "        'Time' = sfit$time,\n",
    "        'surv' = sfit$surv,\n",
    "        'Event' = sfit$n.censor,\n",
    "        'lower' = sfit$lower,\n",
    "        'upper' = sfit$upper,\n",
    "        'Group' = strata,\n",
    "        'p' = surv_pvalue(sfit, data = cdata_x)$pval,\n",
    "        'p_txt' = surv_pvalue(sfit, data = cdata_x)$pval.txt\n",
    "    )\n",
    "    #plotdf = plotdf[!is.na(plotdf$lower), ]\n",
    "    #stopifnot(all(table(plotdf$Group) > 1))\n",
    "    \n",
    "    survdfs = list(T_ALL=plotdf)\n",
    "    survdfs = survdfs[!is.na(survdfs)]\n",
    "    \n",
    "    ## -- Return\n",
    "    return(survdfs)\n",
    "    \n",
    "}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "running-spider",
   "metadata": {},
   "source": [
    "<br>\n",
    "\n",
    "<br>\n",
    "\n",
    "# Fig 4D & SuppTable11"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "falling-direction",
   "metadata": {},
   "source": [
    "### Split data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "rapid-workstation",
   "metadata": {},
   "outputs": [],
   "source": [
    "cutquantile = 0.3333\n",
    "\n",
    "## -- By ZBTB16\n",
    "\n",
    "scvec_Z = UPenn_dge$logCPM[\"ZBTB16\",]\n",
    "scstrata_Z = rep(NA, length(scvec_Z))\n",
    "scstrata_Z[scvec_Z < quantile(scvec_Z, cutquantile)] = 'Low'\n",
    "scstrata_Z[scvec_Z > quantile(scvec_Z, 1 - cutquantile)] = 'High'\n",
    "\n",
    "## -- Rename\n",
    "scstrata_Z = gsub('High', ' ^ Hi', scstrata_Z)\n",
    "scstrata_Z = gsub('Low', ' ^ Lo', scstrata_Z)\n",
    "names(scstrata_Z) <- names(scvec_Z)\n",
    "\n",
    "UPenn_dge$samples$survival_group <- as.vector(scstrata_Z[UPenn_dge$samples$sample_id])\n",
    "UPenn_dge$samples$ZBTB16_exprs <-  as.vector(UPenn_dge$logCPM[\"ZBTB16\",])\n",
    "\n",
    "\n",
    "## -- By ETP  // exclude Near-ETP & Unknown\n",
    "\n",
    "scstrata_ETP = rep(NA, nrow(UPenn_dge$samples))\n",
    "scstrata_ETP[UPenn_dge$samples$ETP_status==\"ETP\"] = 'ETP'\n",
    "scstrata_ETP[UPenn_dge$samples$ETP_status==\"Non-ETP\"] = 'Non-ETP'\n",
    "names(scstrata_ETP) <- rownames(UPenn_dge$samples)\n",
    "\n",
    "## -- By ETP  // exclude unknown but include Near-ETP!\n",
    "scstrata_ETP_near = rep(NA, nrow(UPenn_dge$samples))\n",
    "scstrata_ETP_near[UPenn_dge$samples$ETP_status==\"ETP\"] = 'ETP'\n",
    "scstrata_ETP_near[UPenn_dge$samples$ETP_status %in% c(\"Non-ETP\", \"Near-ETP\")] = 'Non-ETP'\n",
    "names(scstrata_ETP_near) <- rownames(UPenn_dge$samples)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "weighted-street",
   "metadata": {},
   "source": [
    "### Run cox"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "satisfied-change",
   "metadata": {},
   "outputs": [],
   "source": [
    "testCOX <- function(ZBTB16_vec_x, include_near=FALSE, status_str=\"OS.status\", event_str = \"OS\"){\n",
    "    ### Alternative:\n",
    "    # - status_str = \"EFS.status\"\n",
    "    # - event_str = \"EFS\"\n",
    "    \n",
    "    ## -- Get ETP vector\n",
    "    if (include_near==FALSE){\n",
    "        ETP_vec_x = rep(NA, nrow(UPenn_dge$samples))\n",
    "        ETP_vec_x[UPenn_dge$samples$ETP_status==\"ETP\"] = 'ETP'\n",
    "        ETP_vec_x[UPenn_dge$samples$ETP_status==\"Non-ETP\"] = 'Non-ETP'\n",
    "        names(ETP_vec_x) <- rownames(UPenn_dge$samples)\n",
    "    } else {\n",
    "        ETP_vec_x = rep(NA, nrow(UPenn_dge$samples))\n",
    "        ETP_vec_x[UPenn_dge$samples$ETP_status==\"ETP\"] = 'ETP'\n",
    "        ETP_vec_x[UPenn_dge$samples$ETP_status %in% c(\"Non-ETP\", \"Near-ETP\")] = 'Non-ETP'\n",
    "        names(ETP_vec_x) <- rownames(UPenn_dge$samples)\n",
    "    }\n",
    "\n",
    "    \n",
    "    ## -- Prep\n",
    "    cdata_x = UPenn_dge$samples[rownames(UPenn_dge$samples) %in% names(ETP_vec_x[!(is.na(ETP_vec_x))]),]\n",
    "\n",
    "\n",
    "\n",
    "    ## -- Check values\n",
    "    if (length(cdata_x[which((cdata_x[, event_str] < 0) == TRUE), \n",
    "                   \"short_name\"]) > 0 ){\n",
    "            print(\"Some patients have negative values for event\")\n",
    "    }\n",
    "    \n",
    "    # -- Create into survival data structure\n",
    "    cdata_x$s = grepl(\"1\", cdata_x[[status_str]], ignore.case = TRUE)\n",
    "    cdata_x$ETP = factor(as.vector(ETP_vec_x[rownames(cdata_x)]), levels=unique(as.vector(ETP_vec_x[rownames(cdata_x)])))\n",
    "\n",
    "    if (is.character(ZBTB16_vec_x[[1]])){\n",
    "        cdata_x$Z_expres <- factor(as.vector(ZBTB16_vec_x[rownames(cdata_x)]))\n",
    "        cdata_x$Z_expres <- relevel(cdata_x$Z_expres, \" ^ Lo\")\n",
    "    } else {\n",
    "        cdata_x$Z_expres = as.vector(ZBTB16_vec_x[rownames(cdata_x)])\n",
    "    }\n",
    "\n",
    "    cdata_x$event <- as.vector(cdata_x[[event_str]])\n",
    "    cdata_x = cdata_x[, c(\"event\", \"s\", \"ETP\",\"Z_expres\")]\n",
    "    cdata_x = cdata_x[order(cdata_x$ETP),]\n",
    "\n",
    "    #if (class(cdata_x$Z_expres)==\"numeric\"){\n",
    "    #    Z_str = \"Z_expres\"\n",
    "    #} else {\n",
    "    #    cdata_x$Z_expres <- as.factor(cdata_x$Z_expres)\n",
    "    #    cdata_x = cdata_x[rev(order(cdata_x$Z_expres)),]\n",
    "    #    Z_str = paste0(\"Z_expres\",unique(cdata_x$Z_expres[!(is.na(cdata_x$Z_expres))])[2])    \n",
    "    #}\n",
    "    \n",
    "    ## -- Set formula\n",
    "    f = as.formula(\"Surv(event, s) ~ ETP + Z_expres\")\n",
    "    \n",
    "    sfit = do.call(survival::coxph, list(formula = f, data = cdata_x))\n",
    "    \n",
    "    res.cox <- coxph(Surv(event, s) ~ ETP + Z_expres, data =  cdata_x,na.action=na.exclude)\n",
    "    \n",
    "   # print(Z_str)\n",
    "    print(summary(res.cox))\n",
    "    \n",
    "\n",
    "    p_x = summary(res.cox)$coefficients[,5][[2]]\n",
    "    p_y = summary(res.cox)$coefficients[,5][[1]]\n",
    "    coef_x = summary(res.cox)$conf.int[,1][[2]]\n",
    "    coef_ETP_x = summary(res.cox)$conf.int[,1][[\"ETPETP\"]]\n",
    "\n",
    "    lower_x  = summary(res.cox)$conf.int[,3][[2]]\n",
    "    upper_x = summary(res.cox)$conf.int[,4][[2]]\n",
    "    lower_ETP_x = summary(res.cox)$conf.int[,3][[\"ETPETP\"]]\n",
    "    upper_ETP_x = summary(res.cox)$conf.int[,4][[\"ETPETP\"]]\n",
    "\n",
    "    signif_x = 3\n",
    "    return(c(signif(p_x, signif_x), signif(p_y, signif_x), \n",
    "             signif(coef_x, signif_x), signif(coef_ETP_x, signif_x),\n",
    "             signif(lower_x, signif_x),signif(upper_x, signif_x),\n",
    "             signif(lower_ETP_x, signif_x), signif(upper_ETP_x, signif_x)))\n",
    "    \n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "featured-bernard",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Call:\n",
      "coxph(formula = Surv(event, s) ~ ETP + Z_expres, data = cdata_x, \n",
      "    na.action = na.exclude)\n",
      "\n",
      "  n= 581, number of events= 72 \n",
      "   (307 observations deleted due to missingness)\n",
      "\n",
      "                coef exp(coef) se(coef)     z Pr(>|z|)    \n",
      "ETPETP        0.1513    1.1634   0.3008 0.503 0.614843    \n",
      "Z_expres ^ Hi 1.0176    2.7667   0.2683 3.793 0.000149 ***\n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "              exp(coef) exp(-coef) lower .95 upper .95\n",
      "ETPETP            1.163     0.8596    0.6452     2.098\n",
      "Z_expres ^ Hi     2.767     0.3614    1.6353     4.681\n",
      "\n",
      "Concordance= 0.629  (se = 0.029 )\n",
      "Likelihood ratio test= 19.4  on 2 df,   p=6e-05\n",
      "Wald test            = 17.91  on 2 df,   p=1e-04\n",
      "Score (logrank) test = 19.67  on 2 df,   p=5e-05\n",
      "\n",
      "Call:\n",
      "coxph(formula = Surv(event, s) ~ ETP + Z_expres, data = cdata_x, \n",
      "    na.action = na.exclude)\n",
      "\n",
      "  n= 697, number of events= 82 \n",
      "   (342 observations deleted due to missingness)\n",
      "\n",
      "                coef exp(coef) se(coef)     z Pr(>|z|)    \n",
      "ETPETP        0.3239    1.3825   0.2904 1.115 0.264780    \n",
      "Z_expres ^ Hi 0.8787    2.4077   0.2530 3.473 0.000515 ***\n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "              exp(coef) exp(-coef) lower .95 upper .95\n",
      "ETPETP            1.382     0.7233    0.7824     2.443\n",
      "Z_expres ^ Hi     2.408     0.4153    1.4664     3.953\n",
      "\n",
      "Concordance= 0.619  (se = 0.028 )\n",
      "Likelihood ratio test= 18.07  on 2 df,   p=1e-04\n",
      "Wald test            = 16.82  on 2 df,   p=2e-04\n",
      "Score (logrank) test = 18.2  on 2 df,   p=1e-04\n",
      "\n",
      "Call:\n",
      "coxph(formula = Surv(event, s) ~ ETP + Z_expres, data = cdata_x, \n",
      "    na.action = na.exclude)\n",
      "\n",
      "  n= 888, number of events= 100 \n",
      "\n",
      "            coef exp(coef) se(coef)     z Pr(>|z|)    \n",
      "ETPETP   0.13100   1.13997  0.28424 0.461    0.645    \n",
      "Z_expres 0.15981   1.17329  0.03835 4.167 3.09e-05 ***\n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "         exp(coef) exp(-coef) lower .95 upper .95\n",
      "ETPETP       1.140     0.8772    0.6531     1.990\n",
      "Z_expres     1.173     0.8523    1.0883     1.265\n",
      "\n",
      "Concordance= 0.627  (se = 0.028 )\n",
      "Likelihood ratio test= 21.61  on 2 df,   p=2e-05\n",
      "Wald test            = 22.79  on 2 df,   p=1e-05\n",
      "Score (logrank) test = 23.6  on 2 df,   p=7e-06\n",
      "\n",
      "Call:\n",
      "coxph(formula = Surv(event, s) ~ ETP + Z_expres, data = cdata_x, \n",
      "    na.action = na.exclude)\n",
      "\n",
      "  n= 1039, number of events= 112 \n",
      "\n",
      "            coef exp(coef) se(coef)     z Pr(>|z|)    \n",
      "ETPETP   0.31260   1.36698  0.27173 1.150 0.249972    \n",
      "Z_expres 0.13199   1.14109  0.03441 3.836 0.000125 ***\n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "         exp(coef) exp(-coef) lower .95 upper .95\n",
      "ETPETP       1.367     0.7315    0.8025     2.328\n",
      "Z_expres     1.141     0.8764    1.0667     1.221\n",
      "\n",
      "Concordance= 0.62  (se = 0.026 )\n",
      "Likelihood ratio test= 20.04  on 2 df,   p=4e-05\n",
      "Wald test            = 21.05  on 2 df,   p=3e-05\n",
      "Score (logrank) test = 21.82  on 2 df,   p=2e-05\n",
      "\n",
      "Call:\n",
      "coxph(formula = Surv(event, s) ~ ETP + Z_expres, data = cdata_x, \n",
      "    na.action = na.exclude)\n",
      "\n",
      "  n= 581, number of events= 100 \n",
      "   (307 observations deleted due to missingness)\n",
      "\n",
      "                coef exp(coef) se(coef)     z Pr(>|z|)   \n",
      "ETPETP        0.2275    1.2554   0.2642 0.861   0.3892   \n",
      "Z_expres ^ Hi 0.7202    2.0549   0.2207 3.263   0.0011 **\n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "              exp(coef) exp(-coef) lower .95 upper .95\n",
      "ETPETP            1.255     0.7965     0.748     2.107\n",
      "Z_expres ^ Hi     2.055     0.4866     1.333     3.167\n",
      "\n",
      "Concordance= 0.608  (se = 0.026 )\n",
      "Likelihood ratio test= 15.86  on 2 df,   p=4e-04\n",
      "Wald test            = 15.59  on 2 df,   p=4e-04\n",
      "Score (logrank) test = 16.45  on 2 df,   p=3e-04\n",
      "\n",
      "Call:\n",
      "coxph(formula = Surv(event, s) ~ ETP + Z_expres, data = cdata_x, \n",
      "    na.action = na.exclude)\n",
      "\n",
      "  n= 697, number of events= 122 \n",
      "   (342 observations deleted due to missingness)\n",
      "\n",
      "                coef exp(coef) se(coef)     z Pr(>|z|)    \n",
      "ETPETP        0.2388    1.2698   0.2507 0.953  0.34069    \n",
      "Z_expres ^ Hi 0.7015    2.0167   0.1992 3.521  0.00043 ***\n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "              exp(coef) exp(-coef) lower .95 upper .95\n",
      "ETPETP            1.270     0.7876    0.7769     2.075\n",
      "Z_expres ^ Hi     2.017     0.4959    1.3648     2.980\n",
      "\n",
      "Concordance= 0.604  (se = 0.023 )\n",
      "Likelihood ratio test= 17.33  on 2 df,   p=2e-04\n",
      "Wald test            = 16.66  on 2 df,   p=2e-04\n",
      "Score (logrank) test = 17.51  on 2 df,   p=2e-04\n",
      "\n",
      "Call:\n",
      "coxph(formula = Surv(event, s) ~ ETP + Z_expres, data = cdata_x, \n",
      "    na.action = na.exclude)\n",
      "\n",
      "  n= 888, number of events= 142 \n",
      "\n",
      "            coef exp(coef) se(coef)     z Pr(>|z|)    \n",
      "ETPETP   0.14253   1.15319  0.24938 0.572 0.567648    \n",
      "Z_expres 0.11996   1.12745  0.03289 3.647 0.000265 ***\n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "         exp(coef) exp(-coef) lower .95 upper .95\n",
      "ETPETP       1.153     0.8672    0.7073     1.880\n",
      "Z_expres     1.127     0.8870    1.0571     1.203\n",
      "\n",
      "Concordance= 0.599  (se = 0.025 )\n",
      "Likelihood ratio test= 17.37  on 2 df,   p=2e-04\n",
      "Wald test            = 18.37  on 2 df,   p=1e-04\n",
      "Score (logrank) test = 18.74  on 2 df,   p=9e-05\n",
      "\n",
      "Call:\n",
      "coxph(formula = Surv(event, s) ~ ETP + Z_expres, data = cdata_x, \n",
      "    na.action = na.exclude)\n",
      "\n",
      "  n= 1039, number of events= 167 \n",
      "\n",
      "            coef exp(coef) se(coef)     z Pr(>|z|)    \n",
      "ETPETP   0.13573   1.14537  0.23675 0.573    0.566    \n",
      "Z_expres 0.13072   1.13964  0.02857 4.575 4.76e-06 ***\n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "         exp(coef) exp(-coef) lower .95 upper .95\n",
      "ETPETP       1.145     0.8731    0.7202     1.822\n",
      "Z_expres     1.140     0.8775    1.0776     1.205\n",
      "\n",
      "Concordance= 0.609  (se = 0.023 )\n",
      "Likelihood ratio test= 24.51  on 2 df,   p=5e-06\n",
      "Wald test            = 25.48  on 2 df,   p=3e-06\n",
      "Score (logrank) test = 26.1  on 2 df,   p=2e-06\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## -- OS\n",
    "vec_1 = testCOX(scstrata_Z, include_near=FALSE, status_str=\"OS.status\", event_str = \"OS\")\n",
    "vec_2 = testCOX(scstrata_Z, include_near=TRUE, status_str=\"OS.status\", event_str = \"OS\")\n",
    "\n",
    "vec_3 = testCOX(scvec_Z, include_near=FALSE, status_str=\"OS.status\", event_str = \"OS\")\n",
    "vec_4 = testCOX(scvec_Z, include_near=TRUE, status_str=\"OS.status\", event_str = \"OS\")\n",
    "\n",
    "\n",
    "## -- EFS\n",
    "vec_5 = testCOX(scstrata_Z, include_near=FALSE, status_str=\"EFS.status\", event_str = \"EFS\")\n",
    "vec_6 = testCOX(scstrata_Z, include_near=TRUE, status_str=\"EFS.status\", event_str = \"EFS\")\n",
    "\n",
    "vec_7 = testCOX(scvec_Z, include_near=FALSE, status_str=\"EFS.status\", event_str = \"EFS\")\n",
    "vec_8 = testCOX(scvec_Z, include_near=TRUE, status_str=\"EFS.status\", event_str = \"EFS\")\n",
    "\n",
    "\n",
    "cox_df = data.frame(Z_tert = vec_1,\n",
    "           Z_tert_near = vec_2,\n",
    "           Z_cont = vec_3,\n",
    "           Z_cont_near = vec_4,\n",
    "           Z_tert_efs = vec_5,\n",
    "           Z_tert_efs_near = vec_6, \n",
    "           Z_cont_efs = vec_7, \n",
    "           Z_cont_efs_near = vec_8, \n",
    "                    \n",
    "                    row.names = c(\"P-value\", \"P-value (ETP)\", \"Coef. ZBTB16\", \"Coef. ETP\",\n",
    "                                            \"Lower .95 ZBTB16\", \"Higher .95 ZBTB16\",\n",
    "                                             \"Lower .95 ETP\", \"Higher .95 ETP\"))\n",
    "cox_df <- t(cox_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "annual-invention",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 8 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>P-value</th><th scope=col>Coef.</th><th scope=col>Lower .95</th><th scope=col>Upper .95</th><th scope=col>P-value</th><th scope=col>Coef.</th><th scope=col>Lower .95</th><th scope=col>Upper .95</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Z_tert</th><td>1.49e-04</td><td>2.77</td><td>1.64</td><td>4.68</td><td>0.615</td><td>1.16</td><td>0.645</td><td>2.10</td></tr>\n",
       "\t<tr><th scope=row>Z_tert_near</th><td>5.15e-04</td><td>2.41</td><td>1.47</td><td>3.95</td><td>0.265</td><td>1.38</td><td>0.782</td><td>2.44</td></tr>\n",
       "\t<tr><th scope=row>Z_cont</th><td>3.09e-05</td><td>1.17</td><td>1.09</td><td>1.26</td><td>0.645</td><td>1.14</td><td>0.653</td><td>1.99</td></tr>\n",
       "\t<tr><th scope=row>Z_cont_near</th><td>1.25e-04</td><td>1.14</td><td>1.07</td><td>1.22</td><td>0.250</td><td>1.37</td><td>0.803</td><td>2.33</td></tr>\n",
       "\t<tr><th scope=row>Z_tert_efs</th><td>1.10e-03</td><td>2.05</td><td>1.33</td><td>3.17</td><td>0.389</td><td>1.26</td><td>0.748</td><td>2.11</td></tr>\n",
       "\t<tr><th scope=row>Z_tert_efs_near</th><td>4.30e-04</td><td>2.02</td><td>1.36</td><td>2.98</td><td>0.341</td><td>1.27</td><td>0.777</td><td>2.08</td></tr>\n",
       "\t<tr><th scope=row>Z_cont_efs</th><td>2.65e-04</td><td>1.13</td><td>1.06</td><td>1.20</td><td>0.568</td><td>1.15</td><td>0.707</td><td>1.88</td></tr>\n",
       "\t<tr><th scope=row>Z_cont_efs_near</th><td>4.76e-06</td><td>1.14</td><td>1.08</td><td>1.21</td><td>0.566</td><td>1.15</td><td>0.720</td><td>1.82</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 8 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & P-value & Coef. & Lower .95 & Upper .95 & P-value & Coef. & Lower .95 & Upper .95\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tZ\\_tert & 1.49e-04 & 2.77 & 1.64 & 4.68 & 0.615 & 1.16 & 0.645 & 2.10\\\\\n",
       "\tZ\\_tert\\_near & 5.15e-04 & 2.41 & 1.47 & 3.95 & 0.265 & 1.38 & 0.782 & 2.44\\\\\n",
       "\tZ\\_cont & 3.09e-05 & 1.17 & 1.09 & 1.26 & 0.645 & 1.14 & 0.653 & 1.99\\\\\n",
       "\tZ\\_cont\\_near & 1.25e-04 & 1.14 & 1.07 & 1.22 & 0.250 & 1.37 & 0.803 & 2.33\\\\\n",
       "\tZ\\_tert\\_efs & 1.10e-03 & 2.05 & 1.33 & 3.17 & 0.389 & 1.26 & 0.748 & 2.11\\\\\n",
       "\tZ\\_tert\\_efs\\_near & 4.30e-04 & 2.02 & 1.36 & 2.98 & 0.341 & 1.27 & 0.777 & 2.08\\\\\n",
       "\tZ\\_cont\\_efs & 2.65e-04 & 1.13 & 1.06 & 1.20 & 0.568 & 1.15 & 0.707 & 1.88\\\\\n",
       "\tZ\\_cont\\_efs\\_near & 4.76e-06 & 1.14 & 1.08 & 1.21 & 0.566 & 1.15 & 0.720 & 1.82\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 8 × 8\n",
       "\n",
       "| <!--/--> | P-value &lt;dbl&gt; | Coef. &lt;dbl&gt; | Lower .95 &lt;dbl&gt; | Upper .95 &lt;dbl&gt; | P-value &lt;dbl&gt; | Coef. &lt;dbl&gt; | Lower .95 &lt;dbl&gt; | Upper .95 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Z_tert | 1.49e-04 | 2.77 | 1.64 | 4.68 | 0.615 | 1.16 | 0.645 | 2.10 |\n",
       "| Z_tert_near | 5.15e-04 | 2.41 | 1.47 | 3.95 | 0.265 | 1.38 | 0.782 | 2.44 |\n",
       "| Z_cont | 3.09e-05 | 1.17 | 1.09 | 1.26 | 0.645 | 1.14 | 0.653 | 1.99 |\n",
       "| Z_cont_near | 1.25e-04 | 1.14 | 1.07 | 1.22 | 0.250 | 1.37 | 0.803 | 2.33 |\n",
       "| Z_tert_efs | 1.10e-03 | 2.05 | 1.33 | 3.17 | 0.389 | 1.26 | 0.748 | 2.11 |\n",
       "| Z_tert_efs_near | 4.30e-04 | 2.02 | 1.36 | 2.98 | 0.341 | 1.27 | 0.777 | 2.08 |\n",
       "| Z_cont_efs | 2.65e-04 | 1.13 | 1.06 | 1.20 | 0.568 | 1.15 | 0.707 | 1.88 |\n",
       "| Z_cont_efs_near | 4.76e-06 | 1.14 | 1.08 | 1.21 | 0.566 | 1.15 | 0.720 | 1.82 |\n",
       "\n"
      ],
      "text/plain": [
       "                P-value  Coef. Lower .95 Upper .95 P-value Coef. Lower .95\n",
       "Z_tert          1.49e-04 2.77  1.64      4.68      0.615   1.16  0.645    \n",
       "Z_tert_near     5.15e-04 2.41  1.47      3.95      0.265   1.38  0.782    \n",
       "Z_cont          3.09e-05 1.17  1.09      1.26      0.645   1.14  0.653    \n",
       "Z_cont_near     1.25e-04 1.14  1.07      1.22      0.250   1.37  0.803    \n",
       "Z_tert_efs      1.10e-03 2.05  1.33      3.17      0.389   1.26  0.748    \n",
       "Z_tert_efs_near 4.30e-04 2.02  1.36      2.98      0.341   1.27  0.777    \n",
       "Z_cont_efs      2.65e-04 1.13  1.06      1.20      0.568   1.15  0.707    \n",
       "Z_cont_efs_near 4.76e-06 1.14  1.08      1.21      0.566   1.15  0.720    \n",
       "                Upper .95\n",
       "Z_tert          2.10     \n",
       "Z_tert_near     2.44     \n",
       "Z_cont          1.99     \n",
       "Z_cont_near     2.33     \n",
       "Z_tert_efs      2.11     \n",
       "Z_tert_efs_near 2.08     \n",
       "Z_cont_efs      1.88     \n",
       "Z_cont_efs_near 1.82     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "column_ordering <- c(\"P-value\", \"Coef. ZBTB16\", \"Lower .95 ZBTB16\", \"Higher .95 ZBTB16\",\"P-value (ETP)\", \"Coef. ETP\", \"Lower .95 ETP\", \"Higher .95 ETP\")\n",
    "cox_df <- as.data.frame(cox_df[,column_ordering])\n",
    "\n",
    "colnames(cox_df) <- c(\"P-value\", \"Coef.\", \"Lower .95\", \"Upper .95\", \"P-value\", \"Coef.\", \"Lower .95\", \"Upper .95\")\n",
    "\n",
    "cox_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "settled-liberty",
   "metadata": {},
   "source": [
    "### Save"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "selected-parks",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in write.csv(cox_df, paste0(save_dir, \"SuppTable11_and_Fig4D.csv\"), :\n",
      "“attempt to set 'sep' ignored”\n"
     ]
    }
   ],
   "source": [
    "write.csv(cox_df,paste0(save_dir,\"SuppTable11_and_Fig4D.csv\"), sep=\",\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "verified-center",
   "metadata": {},
   "source": [
    "<br>\n",
    "\n",
    "<br>\n",
    "\n",
    "# Fig 4E & SuppTable13"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "stone-passenger",
   "metadata": {},
   "source": [
    "### Set up"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "suburban-ideal",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in checkGenes(upSet, rownames(rankData)):\n",
      "“4 genes missing: L1TD1, PTH2, MTRNR2L8, AC103591.3”\n",
      "Warning message in checkGenes(upSet, rownames(rankData)):\n",
      "“1 genes missing: IL3RA”\n",
      "Warning message in checkGenes(upSet, rownames(rankData)):\n",
      "“2 genes missing: TRBV2, NNMT”\n"
     ]
    }
   ],
   "source": [
    "Z_modules <- getGmt(paste0(\"/lustre/scratch127/recovered/scratch126/lustre/scratch126/casm/team274sb/hw12/T_ALL/modules/\", \n",
    "                     \"ZBTB16_modules_refined.gmt\"))\n",
    "\n",
    "\n",
    "teachy_df = read.table(\"/lustre/scratch127/recovered/scratch126/lustre/scratch126/casm/team274sb/project_folders/T_ALL_paper/data/teachy_sig.csv\", sep=\",\", header=TRUE)\n",
    "teachy_sig = teachy_df$gene\n",
    "\n",
    "teachy_df_17 = read.table(\"/lustre/scratch127/recovered/scratch126/lustre/scratch126/casm/team274sb/project_folders/T_ALL_paper/data/teachy_sig_17.csv\", sep=\",\", header=TRUE)\n",
    "teachy_17_sig = teachy_df_17$gene\n",
    "\n",
    "teachy_9_sig = c(\"CD33\", \"IL3RA\", \"PTPRC\", \"HLA-DRA\", \"CD3E\", \"CD8A\", \"CD2\",\"CD4\",\"CD1A\")\n",
    "\n",
    "\n",
    "\n",
    "### --- Score\n",
    "library(singscore)\n",
    "\n",
    "cpm_data <- UPenn_dge$logCPM\n",
    "\n",
    "## -- Rank genes\n",
    "eranks = rankGenes(cpm_data, stableGenes=getStableGenes(10, type = 'blood'))\n",
    "#eranks = rankGenes(cpm_data)\n",
    "\n",
    "## -- Score genest\n",
    "T_scores_vec = simpleScore(eranks, upSet=teachy_sig, centerScore=FALSE)$TotalScore\n",
    "T17_scores_vec = simpleScore(eranks, upSet=teachy_17_sig, centerScore=FALSE)$TotalScore\n",
    "T9_scores_vec = simpleScore(eranks, upSet=teachy_9_sig, centerScore=FALSE)$TotalScore\n",
    "Z_scores_vec = simpleScore(eranks, upSet=geneIds(Z_modules[[\"refinedES_module_up\"]]), centerScore=FALSE)$TotalScore\n",
    "\n",
    "names(T_scores_vec) <- colnames(eranks)\n",
    "names(T17_scores_vec) <- colnames(eranks)\n",
    "names(T9_scores_vec) <- colnames(eranks)\n",
    "names(Z_scores_vec) <- colnames(eranks)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "gorgeous-lecture",
   "metadata": {},
   "source": [
    "### Split data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "worldwide-alarm",
   "metadata": {},
   "outputs": [],
   "source": [
    "cutquantile = 0.3333\n",
    "## -- By ZBTB16\n",
    "\n",
    "scvec_Z = UPenn_dge$logCPM[\"ZBTB16\",]\n",
    "scstrata_Z = rep(NA, length(scvec_Z))\n",
    "scstrata_Z[scvec_Z < quantile(scvec_Z, cutquantile)] = 'Low'\n",
    "scstrata_Z[scvec_Z > quantile(scvec_Z, 1 - cutquantile)] = 'High'\n",
    "\n",
    "## -- Rename\n",
    "scstrata_Z = gsub('High', ' ^ Hi', scstrata_Z)\n",
    "scstrata_Z = gsub('Low', ' ^ Lo', scstrata_Z)\n",
    "names(scstrata_Z) <- names(scvec_Z)\n",
    "\n",
    "UPenn_dge$samples$survival_group <- as.vector(scstrata_Z[UPenn_dge$samples$sample_id])\n",
    "UPenn_dge$samples$ZBTB16_exprs <-  as.vector(UPenn_dge$logCPM[\"ZBTB16\",])\n",
    "\n",
    "## -- By our module score\n",
    "scvec_Z = Z_scores_vec\n",
    "scstrata_Z = rep(\"Mid\", length(scvec_Z))\n",
    "scstrata_Z[scvec_Z < quantile(scvec_Z, cutquantile)] = 'Low'\n",
    "scstrata_Z[scvec_Z > quantile(scvec_Z, 1 - cutquantile)] = 'High'\n",
    "names(scstrata_Z) <- rownames(UPenn_dge$samples)\n",
    "UPenn_dge$samples$Z_module_group <- as.vector(scstrata_Z[UPenn_dge$samples$sample_id])\n",
    "\n",
    "## -- By their module score\n",
    "\n",
    "scstrata_T = rep(\"Mid\", length(T_scores_vec))\n",
    "scstrata_T[T_scores_vec < quantile(T_scores_vec, cutquantile)] = 'Low'\n",
    "scstrata_T[T_scores_vec > quantile(T_scores_vec, 1 - cutquantile)] = 'High'\n",
    "names(scstrata_T) <- rownames(UPenn_dge$samples)\n",
    "UPenn_dge$samples$T_module_group <- as.vector(scstrata_T[UPenn_dge$samples$sample_id])\n",
    "\n",
    "## -- By their module score -- BMP-17\n",
    "\n",
    "scstrata_T17 = rep(\"Mid\", length(T17_scores_vec))\n",
    "scstrata_T17[T17_scores_vec < quantile(T17_scores_vec, cutquantile)] = 'Low'\n",
    "scstrata_T17[T17_scores_vec > quantile(T17_scores_vec, 1 - cutquantile)] = 'High'\n",
    "names(scstrata_T17) <- rownames(UPenn_dge$samples)\n",
    "UPenn_dge$samples$T_17_module_group <- as.vector(scstrata_T17[UPenn_dge$samples$sample_id])\n",
    "\n",
    "## -- By their module score -- BMP-surface-9\n",
    "scstrata_T9 = rep(\"Mid\", length(T9_scores_vec))\n",
    "scstrata_T9[T9_scores_vec < quantile(T9_scores_vec, cutquantile)] = 'Low'\n",
    "scstrata_T9[T9_scores_vec > quantile(T9_scores_vec, 1 - cutquantile)] = 'High'\n",
    "names(scstrata_T9) <- rownames(UPenn_dge$samples)\n",
    "UPenn_dge$samples$T_9_module_group <- as.vector(scstrata_T9[UPenn_dge$samples$sample_id])\n",
    "\n",
    "\n",
    "## -- By ETP  // exclude Near-ETP & Unknown\n",
    "\n",
    "scstrata_ETP = rep(NA, nrow(UPenn_dge$samples))\n",
    "scstrata_ETP[UPenn_dge$samples$ETP_status==\"ETP\"] = 'ETP'\n",
    "scstrata_ETP[UPenn_dge$samples$ETP_status==\"Non-ETP\"] = 'Non-ETP'\n",
    "names(scstrata_ETP) <- rownames(UPenn_dge$samples)\n",
    "\n",
    "## -- By ETP  // exclude unknown but include Near-ETP!\n",
    "scstrata_ETP_near = rep(NA, nrow(UPenn_dge$samples))\n",
    "scstrata_ETP_near[UPenn_dge$samples$ETP_status==\"ETP\"] = 'ETP'\n",
    "scstrata_ETP_near[UPenn_dge$samples$ETP_status %in% c(\"Non-ETP\", \"Near-ETP\")] = 'Non-ETP'\n",
    "names(scstrata_ETP_near) <- rownames(UPenn_dge$samples)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "minimal-oxford",
   "metadata": {},
   "source": [
    "### Run cox"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "employed-shirt",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### ---- FUNCTIONS\n",
    "do_cox <- function(x_str, y_str, include_ETP=FALSE, status_str=\"OS.status\", event_str = \"OS\"){\n",
    "    \n",
    "    # -- Create into survival data structure\n",
    "    cdata_x$s = grepl(\"1\", cdata_x[[status_str]], ignore.case = TRUE)\n",
    "    cdata_x$event <- as.vector(cdata_x[[event_str]])\n",
    "    cdata_x = cdata_x[, c(\"event\", \"s\", \"Z_cont\", \"Z_expres\",\"T_cont\", \"T17_cont\", \"T_cont_s\", \"T17_cont_s\", \"Z_cont_s\", \"T9_cont_s\", \"T9_cont\", \"ETP\")]\n",
    "    \n",
    "    cdata_x$x <- cdata_x[[x_str]]\n",
    "    cdata_x$y <- cdata_x[[y_str]]\n",
    "    \n",
    "    ## -- Run model\n",
    "    if (include_ETP){\n",
    "        cdata_x$ETP <- relevel(cdata_x$ETP, \"Non-ETP\")\n",
    "        res.cox <- coxph(Surv(event, s) ~ x + y + ETP, data =  cdata_x,na.action=na.exclude)\n",
    "    } else {\n",
    "        res.cox <- coxph(Surv(event, s) ~ x + y, data =  cdata_x,na.action=na.exclude)\n",
    "    }\n",
    "    print(paste0(\"X = \",x_str))\n",
    "    print(paste0(\"Y = \",y_str))\n",
    "    print(summary(res.cox)) \n",
    "    \n",
    "    return(res.cox)\n",
    "\n",
    "}\n",
    "\n",
    "get_vec <- function(model_x, str_x){\n",
    "     signif(summary(model_x)$coefficients[,5][[str_x]], 3)\n",
    "    \n",
    "    return(c(c(signif(summary(model_x)$conf.int[,1][[str_x]], 3), \n",
    "               signif(summary(model_x)$conf.int[,3][[str_x]], 3),  \n",
    "               signif(summary(model_x)$conf.int[,4][[str_x]], 3),\n",
    "              signif(summary(model_x)$coefficients[,5][[str_x]], 3))))\n",
    "}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "particular-sword",
   "metadata": {},
   "source": [
    "### Prep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "experienced-publisher",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Not factors because continuous\n",
    "UPenn_dge$samples$T_cont <- as.vector(T_scores_vec[rownames(UPenn_dge$sample)])\n",
    "UPenn_dge$samples$T17_cont <- as.vector(T17_scores_vec[rownames(UPenn_dge$sample)])\n",
    "UPenn_dge$samples$T9_cont <- as.vector(T9_scores_vec[rownames(UPenn_dge$sample)])\n",
    "UPenn_dge$samples$Z_cont <- as.vector(Z_scores_vec[rownames(UPenn_dge$sample)])\n",
    "\n",
    "UPenn_dge$samples$T_cont_s <- scale(as.vector(T_scores_vec[rownames(UPenn_dge$sample)]))\n",
    "UPenn_dge$samples$T17_cont_s <- scale(as.vector(T17_scores_vec[rownames(UPenn_dge$sample)]))\n",
    "UPenn_dge$samples$T9_cont_s <- scale(as.vector(T9_scores_vec[rownames(UPenn_dge$sample)]))\n",
    "UPenn_dge$samples$Z_cont_s <- scale(as.vector(Z_scores_vec[rownames(UPenn_dge$sample)]))\n",
    "\n",
    "UPenn_dge$samples$Z_expres <- as.vector(UPenn_dge$logCPM[\"ZBTB16\",][rownames(UPenn_dge$sample)])\n",
    "UPenn_dge$samples$ETP <- factor(scstrata_ETP_near[rownames(UPenn_dge$sample)])\n",
    "\n",
    "cdata_x = UPenn_dge$samples"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "executed-phoenix",
   "metadata": {},
   "source": [
    "#### OS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "exposed-adjustment",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"X = Z_cont_s\"\n",
      "[1] \"Y = T_cont_s\"\n",
      "Call:\n",
      "coxph(formula = Surv(event, s) ~ x + y, data = cdata_x, na.action = na.exclude)\n",
      "\n",
      "  n= 1175, number of events= 127 \n",
      "\n",
      "     coef exp(coef) se(coef)     z Pr(>|z|)    \n",
      "x 0.36953   1.44706  0.08956 4.126 3.69e-05 ***\n",
      "y 0.11519   1.12209  0.10059 1.145    0.252    \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "  exp(coef) exp(-coef) lower .95 upper .95\n",
      "x     1.447     0.6911    1.2141     1.725\n",
      "y     1.122     0.8912    0.9213     1.367\n",
      "\n",
      "Concordance= 0.635  (se = 0.024 )\n",
      "Likelihood ratio test= 29.84  on 2 df,   p=3e-07\n",
      "Wald test            = 32.91  on 2 df,   p=7e-08\n",
      "Score (logrank) test = 34.36  on 2 df,   p=3e-08\n",
      "\n",
      "[1] \"X = Z_cont_s\"\n",
      "[1] \"Y = T17_cont_s\"\n",
      "Call:\n",
      "coxph(formula = Surv(event, s) ~ x + y, data = cdata_x, na.action = na.exclude)\n",
      "\n",
      "  n= 1175, number of events= 127 \n",
      "\n",
      "    coef exp(coef) se(coef)     z Pr(>|z|)   \n",
      "x 0.3074    1.3599   0.1120 2.746  0.00604 **\n",
      "y 0.1680    1.1830   0.1171 1.435  0.15115   \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "  exp(coef) exp(-coef) lower .95 upper .95\n",
      "x     1.360     0.7353    1.0920     1.694\n",
      "y     1.183     0.8453    0.9405     1.488\n",
      "\n",
      "Concordance= 0.636  (se = 0.024 )\n",
      "Likelihood ratio test= 30.57  on 2 df,   p=2e-07\n",
      "Wald test            = 33.72  on 2 df,   p=5e-08\n",
      "Score (logrank) test = 35.36  on 2 df,   p=2e-08\n",
      "\n",
      "[1] \"X = Z_cont_s\"\n",
      "[1] \"Y = T9_cont_s\"\n",
      "Call:\n",
      "coxph(formula = Surv(event, s) ~ x + y, data = cdata_x, na.action = na.exclude)\n",
      "\n",
      "  n= 1175, number of events= 127 \n",
      "\n",
      "      coef exp(coef) se(coef)      z Pr(>|z|)    \n",
      "x  0.42098   1.52346  0.07689  5.475 4.37e-08 ***\n",
      "y -0.01627   0.98386  0.09309 -0.175    0.861    \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "  exp(coef) exp(-coef) lower .95 upper .95\n",
      "x    1.5235     0.6564    1.3103     1.771\n",
      "y    0.9839     1.0164    0.8198     1.181\n",
      "\n",
      "Concordance= 0.63  (se = 0.024 )\n",
      "Likelihood ratio test= 28.56  on 2 df,   p=6e-07\n",
      "Wald test            = 32.08  on 2 df,   p=1e-07\n",
      "Score (logrank) test = 33.42  on 2 df,   p=6e-08\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## -- OS\n",
    "\n",
    "model_1 <- do_cox(\"Z_cont_s\", \"T_cont_s\",include_ETP=FALSE)\n",
    "#model_2 <- do_cox(\"Z_expres\", \"T_cont_s\",include_ETP=FALSE)\n",
    "\n",
    "model_2 <- do_cox(\"Z_cont_s\", \"T17_cont_s\",include_ETP=FALSE)\n",
    "#model_4 <- do_cox(\"Z_expres\", \"T17_cont_s\",include_ETP=FALSE)\n",
    "\n",
    "model_3 <- do_cox(\"Z_cont_s\", \"T9_cont_s\",include_ETP=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "lightweight-second",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Model</th><th scope=col>Variable</th><th scope=col>Coef.</th><th scope=col>Lower .95</th><th scope=col>Higher .95</th><th scope=col>P-value</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>m1_Z</th><td>m1</td><td>Module       </td><td>1.450</td><td>1.210</td><td>1.72</td><td>3.69e-05</td></tr>\n",
       "\t<tr><th scope=row>m1_T</th><td>m1</td><td>BMP-119      </td><td>1.120</td><td>0.921</td><td>1.37</td><td>2.52e-01</td></tr>\n",
       "\t<tr><th scope=row>m3_Z</th><td>m3</td><td>Module       </td><td>1.360</td><td>1.090</td><td>1.69</td><td>6.04e-03</td></tr>\n",
       "\t<tr><th scope=row>m3_T17</th><td>m3</td><td>BMP-17       </td><td>1.180</td><td>0.940</td><td>1.49</td><td>1.51e-01</td></tr>\n",
       "\t<tr><th scope=row>m5_Z</th><td>m5</td><td>Module       </td><td>1.520</td><td>1.310</td><td>1.77</td><td>4.37e-08</td></tr>\n",
       "\t<tr><th scope=row>m5_T9</th><td>m5</td><td>BMP-surface-9</td><td>0.984</td><td>0.820</td><td>1.18</td><td>8.61e-01</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & Model & Variable & Coef. & Lower .95 & Higher .95 & P-value\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tm1\\_Z & m1 & Module        & 1.450 & 1.210 & 1.72 & 3.69e-05\\\\\n",
       "\tm1\\_T & m1 & BMP-119       & 1.120 & 0.921 & 1.37 & 2.52e-01\\\\\n",
       "\tm3\\_Z & m3 & Module        & 1.360 & 1.090 & 1.69 & 6.04e-03\\\\\n",
       "\tm3\\_T17 & m3 & BMP-17        & 1.180 & 0.940 & 1.49 & 1.51e-01\\\\\n",
       "\tm5\\_Z & m5 & Module        & 1.520 & 1.310 & 1.77 & 4.37e-08\\\\\n",
       "\tm5\\_T9 & m5 & BMP-surface-9 & 0.984 & 0.820 & 1.18 & 8.61e-01\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | Model &lt;chr&gt; | Variable &lt;chr&gt; | Coef. &lt;dbl&gt; | Lower .95 &lt;dbl&gt; | Higher .95 &lt;dbl&gt; | P-value &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| m1_Z | m1 | Module        | 1.450 | 1.210 | 1.72 | 3.69e-05 |\n",
       "| m1_T | m1 | BMP-119       | 1.120 | 0.921 | 1.37 | 2.52e-01 |\n",
       "| m3_Z | m3 | Module        | 1.360 | 1.090 | 1.69 | 6.04e-03 |\n",
       "| m3_T17 | m3 | BMP-17        | 1.180 | 0.940 | 1.49 | 1.51e-01 |\n",
       "| m5_Z | m5 | Module        | 1.520 | 1.310 | 1.77 | 4.37e-08 |\n",
       "| m5_T9 | m5 | BMP-surface-9 | 0.984 | 0.820 | 1.18 | 8.61e-01 |\n",
       "\n"
      ],
      "text/plain": [
       "       Model Variable      Coef. Lower .95 Higher .95 P-value \n",
       "m1_Z   m1    Module        1.450 1.210     1.72       3.69e-05\n",
       "m1_T   m1    BMP-119       1.120 0.921     1.37       2.52e-01\n",
       "m3_Z   m3    Module        1.360 1.090     1.69       6.04e-03\n",
       "m3_T17 m3    BMP-17        1.180 0.940     1.49       1.51e-01\n",
       "m5_Z   m5    Module        1.520 1.310     1.77       4.37e-08\n",
       "m5_T9  m5    BMP-surface-9 0.984 0.820     1.18       8.61e-01"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### -- make dataframe\n",
    "\n",
    "cox_df = data.frame(m1_Z = get_vec(model_1, \"x\"),\n",
    "                    m1_T = get_vec(model_1, \"y\"),\n",
    "                #    m1_ETP = get_vec(model_1, \"ETPETP\"),\n",
    "                    \n",
    "                #    m2_Zexpres = get_vec(model_2, \"x\"),\n",
    "                #    m2_T = get_vec(model_2, \"y\"),\n",
    "                #    m2_ETP = get_vec(model_2, \"ETPETP\"),\n",
    "                    \n",
    "                    m3_Z = get_vec(model_2, \"x\"),\n",
    "                    m3_T17 = get_vec(model_2, \"y\"),\n",
    "                #    m3_ETP = get_vec(model_3, \"ETPETP\"),\n",
    "                    \n",
    "               #     m4_Zexpres = get_vec(model_4, \"x\"),\n",
    "               #     m4_T17 = get_vec(model_4, \"y\"),\n",
    "                #    m4_ETP = get_vec(model_4, \"ETPETP\"),\n",
    "                    \n",
    "                    m5_Z = get_vec(model_3, \"x\"),\n",
    "                    m5_T9 = get_vec(model_3, \"y\"),\n",
    "                #    m5_ETP = get_vec(model_5, \"ETPETP\"),\n",
    "                    \n",
    "               #     m6_Zexpres = get_vec(model_6, \"x\"),\n",
    "               #     m6_T9 = get_vec(model_6, \"y\"),\n",
    "                  #  m6_ETP = get_vec(model_6, \"ETPETP\"),\n",
    "                    \n",
    "                    \n",
    "                    row.names = c(\"Coef.\", \"Lower .95\", \"Higher .95\", \"P-value\"))\n",
    "cox_df <- as.data.frame(t(cox_df))\n",
    "\n",
    "cox_df$Model <- unlist(lapply(rownames(cox_df), function(x){strsplit(x,\"_\")[[1]][1]}))\n",
    "\n",
    "var_dict <- setNames(c(\"Module\", \"ZBTB16\", \"BMP-119\", \"BMP-17\", \"BMP-surface-9\",\"ETP\"), c(\"Z\", \"Zexpres\", \"T\", \"T17\", \"T9\", \"ETP\"))\n",
    "cox_df$Variable <- unlist(lapply(rownames(cox_df), function(x){var_dict[[strsplit(x,\"_\")[[1]][2]]]}))\n",
    "\n",
    "column_ordering <- c(\"Model\", \"Variable\", \"Coef.\", \"Lower .95\", \"Higher .95\", \"P-value\")\n",
    "cox_df <- cox_df[,column_ordering]\n",
    "\n",
    "cox_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "continuing-coffee",
   "metadata": {},
   "source": [
    "#### Save"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "color-reservoir",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in write.csv(cox_df, paste0(save_dir, \"SuppTable13_OS.csv\"), sep = \",\"):\n",
      "“attempt to set 'sep' ignored”\n"
     ]
    }
   ],
   "source": [
    "write.csv(cox_df,paste0(save_dir,\"SuppTable13_OS.csv\"), sep=\",\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "compact-going",
   "metadata": {},
   "source": [
    "<br>\n",
    "\n",
    "#### EFS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "played-collective",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"X = Z_cont_s\"\n",
      "[1] \"Y = T_cont_s\"\n",
      "Call:\n",
      "coxph(formula = Surv(event, s) ~ x + y, data = cdata_x, na.action = na.exclude)\n",
      "\n",
      "  n= 1175, number of events= 194 \n",
      "\n",
      "     coef exp(coef) se(coef)     z Pr(>|z|)    \n",
      "x 0.40127   1.49372  0.07400 5.423 5.86e-08 ***\n",
      "y 0.09457   1.09919  0.08219 1.151     0.25    \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "  exp(coef) exp(-coef) lower .95 upper .95\n",
      "x     1.494     0.6695    1.2921     1.727\n",
      "y     1.099     0.9098    0.9357     1.291\n",
      "\n",
      "Concordance= 0.635  (se = 0.021 )\n",
      "Likelihood ratio test= 47.77  on 2 df,   p=4e-11\n",
      "Wald test            = 53.5  on 2 df,   p=2e-12\n",
      "Score (logrank) test = 55.76  on 2 df,   p=8e-13\n",
      "\n",
      "[1] \"X = Z_cont_s\"\n",
      "[1] \"Y = T17_cont_s\"\n",
      "Call:\n",
      "coxph(formula = Surv(event, s) ~ x + y, data = cdata_x, na.action = na.exclude)\n",
      "\n",
      "  n= 1175, number of events= 194 \n",
      "\n",
      "     coef exp(coef) se(coef)     z Pr(>|z|)    \n",
      "x 0.36998   1.44771  0.09206 4.019 5.85e-05 ***\n",
      "y 0.11091   1.11730  0.09637 1.151     0.25    \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "  exp(coef) exp(-coef) lower .95 upper .95\n",
      "x     1.448     0.6907     1.209     1.734\n",
      "y     1.117     0.8950     0.925     1.350\n",
      "\n",
      "Concordance= 0.634  (se = 0.021 )\n",
      "Likelihood ratio test= 47.76  on 2 df,   p=4e-11\n",
      "Wald test            = 53.53  on 2 df,   p=2e-12\n",
      "Score (logrank) test = 56.02  on 2 df,   p=7e-13\n",
      "\n",
      "[1] \"X = Z_cont_s\"\n",
      "[1] \"Y = T9_cont_s\"\n",
      "Call:\n",
      "coxph(formula = Surv(event, s) ~ x + y, data = cdata_x, na.action = na.exclude)\n",
      "\n",
      "  n= 1175, number of events= 194 \n",
      "\n",
      "      coef exp(coef) se(coef)      z Pr(>|z|)    \n",
      "x  0.44550   1.56127  0.06282  7.092 1.32e-12 ***\n",
      "y -0.01220   0.98787  0.07463 -0.164     0.87    \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "  exp(coef) exp(-coef) lower .95 upper .95\n",
      "x    1.5613     0.6405    1.3804     1.766\n",
      "y    0.9879     1.0123    0.8534     1.143\n",
      "\n",
      "Concordance= 0.63  (se = 0.021 )\n",
      "Likelihood ratio test= 46.47  on 2 df,   p=8e-11\n",
      "Wald test            = 52.71  on 2 df,   p=4e-12\n",
      "Score (logrank) test = 54.87  on 2 df,   p=1e-12\n",
      "\n"
     ]
    }
   ],
   "source": [
    "model_4 <- do_cox(\"Z_cont_s\", \"T_cont_s\",include_ETP=FALSE, status_str=\"EFS.status\", event_str = \"EFS\")\n",
    "#model_2 <- do_cox(\"Z_expres\", \"T_cont_s\",include_ETP=FALSE, status_str=\"EFS.status\", event_str = \"EFS\")\n",
    "\n",
    "model_5 <- do_cox(\"Z_cont_s\", \"T17_cont_s\",include_ETP=FALSE, status_str=\"EFS.status\", event_str = \"EFS\")\n",
    "#model_4 <- do_cox(\"Z_expres\", \"T17_cont_s\",include_ETP=FALSE, status_str=\"EFS.status\", event_str = \"EFS\")\n",
    "\n",
    "model_6 <- do_cox(\"Z_cont_s\", \"T9_cont_s\",include_ETP=FALSE, status_str=\"EFS.status\", event_str = \"EFS\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "manual-character",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Model</th><th scope=col>Variable</th><th scope=col>Coef.</th><th scope=col>Lower .95</th><th scope=col>Higher .95</th><th scope=col>P-value</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>m1_Z</th><td>m1</td><td>Module       </td><td>1.490</td><td>1.290</td><td>1.73</td><td>5.86e-08</td></tr>\n",
       "\t<tr><th scope=row>m1_T</th><td>m1</td><td>BMP-119      </td><td>1.100</td><td>0.936</td><td>1.29</td><td>2.50e-01</td></tr>\n",
       "\t<tr><th scope=row>m3_Z</th><td>m3</td><td>Module       </td><td>1.450</td><td>1.210</td><td>1.73</td><td>5.85e-05</td></tr>\n",
       "\t<tr><th scope=row>m3_T17</th><td>m3</td><td>BMP-17       </td><td>1.120</td><td>0.925</td><td>1.35</td><td>2.50e-01</td></tr>\n",
       "\t<tr><th scope=row>m5_Z</th><td>m5</td><td>Module       </td><td>1.560</td><td>1.380</td><td>1.77</td><td>1.32e-12</td></tr>\n",
       "\t<tr><th scope=row>m5_T9</th><td>m5</td><td>BMP-surface-9</td><td>0.988</td><td>0.853</td><td>1.14</td><td>8.70e-01</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & Model & Variable & Coef. & Lower .95 & Higher .95 & P-value\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tm1\\_Z & m1 & Module        & 1.490 & 1.290 & 1.73 & 5.86e-08\\\\\n",
       "\tm1\\_T & m1 & BMP-119       & 1.100 & 0.936 & 1.29 & 2.50e-01\\\\\n",
       "\tm3\\_Z & m3 & Module        & 1.450 & 1.210 & 1.73 & 5.85e-05\\\\\n",
       "\tm3\\_T17 & m3 & BMP-17        & 1.120 & 0.925 & 1.35 & 2.50e-01\\\\\n",
       "\tm5\\_Z & m5 & Module        & 1.560 & 1.380 & 1.77 & 1.32e-12\\\\\n",
       "\tm5\\_T9 & m5 & BMP-surface-9 & 0.988 & 0.853 & 1.14 & 8.70e-01\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | Model &lt;chr&gt; | Variable &lt;chr&gt; | Coef. &lt;dbl&gt; | Lower .95 &lt;dbl&gt; | Higher .95 &lt;dbl&gt; | P-value &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| m1_Z | m1 | Module        | 1.490 | 1.290 | 1.73 | 5.86e-08 |\n",
       "| m1_T | m1 | BMP-119       | 1.100 | 0.936 | 1.29 | 2.50e-01 |\n",
       "| m3_Z | m3 | Module        | 1.450 | 1.210 | 1.73 | 5.85e-05 |\n",
       "| m3_T17 | m3 | BMP-17        | 1.120 | 0.925 | 1.35 | 2.50e-01 |\n",
       "| m5_Z | m5 | Module        | 1.560 | 1.380 | 1.77 | 1.32e-12 |\n",
       "| m5_T9 | m5 | BMP-surface-9 | 0.988 | 0.853 | 1.14 | 8.70e-01 |\n",
       "\n"
      ],
      "text/plain": [
       "       Model Variable      Coef. Lower .95 Higher .95 P-value \n",
       "m1_Z   m1    Module        1.490 1.290     1.73       5.86e-08\n",
       "m1_T   m1    BMP-119       1.100 0.936     1.29       2.50e-01\n",
       "m3_Z   m3    Module        1.450 1.210     1.73       5.85e-05\n",
       "m3_T17 m3    BMP-17        1.120 0.925     1.35       2.50e-01\n",
       "m5_Z   m5    Module        1.560 1.380     1.77       1.32e-12\n",
       "m5_T9  m5    BMP-surface-9 0.988 0.853     1.14       8.70e-01"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### -- make dataframe\n",
    "\n",
    "cox_df = data.frame(m1_Z = get_vec(model_4, \"x\"),\n",
    "                    m1_T = get_vec(model_4, \"y\"),\n",
    "                #    m1_ETP = get_vec(model_1, \"ETPETP\"),\n",
    "                    \n",
    "              #      m2_Zexpres = get_vec(model_2, \"x\"),\n",
    "               #     m2_T = get_vec(model_2, \"y\"),\n",
    "                #    m2_ETP = get_vec(model_2, \"ETPETP\"),\n",
    "                    \n",
    "                    m3_Z = get_vec(model_5, \"x\"),\n",
    "                    m3_T17 = get_vec(model_5, \"y\"),\n",
    "                #    m3_ETP = get_vec(model_3, \"ETPETP\"),\n",
    "                    \n",
    "               #     m4_Zexpres = get_vec(model_4, \"x\"),\n",
    "               #     m4_T17 = get_vec(model_4, \"y\"),\n",
    "                #    m4_ETP = get_vec(model_4, \"ETPETP\"),\n",
    "                    \n",
    "                    m5_Z = get_vec(model_6, \"x\"),\n",
    "                    m5_T9 = get_vec(model_6, \"y\"),\n",
    "                #    m5_ETP = get_vec(model_5, \"ETPETP\"),\n",
    "                    \n",
    "                #    m6_Zexpres = get_vec(model_6, \"x\"),\n",
    "                #    m6_T9 = get_vec(model_6, \"y\"),\n",
    "                  #  m6_ETP = get_vec(model_6, \"ETPETP\"),\n",
    "                    \n",
    "                    \n",
    "                    row.names = c(\"Coef.\", \"Lower .95\", \"Higher .95\", \"P-value\"))\n",
    "cox_df <- as.data.frame(t(cox_df))\n",
    "\n",
    "cox_df$Model <- unlist(lapply(rownames(cox_df), function(x){strsplit(x,\"_\")[[1]][1]}))\n",
    "\n",
    "var_dict <- setNames(c(\"Module\", \"ZBTB16\", \"BMP-119\", \"BMP-17\", \"BMP-surface-9\",\"ETP\"), c(\"Z\", \"Zexpres\", \"T\", \"T17\", \"T9\", \"ETP\"))\n",
    "cox_df$Variable <- unlist(lapply(rownames(cox_df), function(x){var_dict[[strsplit(x,\"_\")[[1]][2]]]}))\n",
    "\n",
    "column_ordering <- c(\"Model\", \"Variable\", \"Coef.\", \"Lower .95\", \"Higher .95\", \"P-value\")\n",
    "cox_df <- cox_df[,column_ordering]\n",
    "\n",
    "cox_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "legal-parliament",
   "metadata": {},
   "source": [
    "#### Save"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "direct-fight",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in write.csv(cox_df, paste0(save_dir, \"SuppTable13_EFS.csv\"), sep = \",\"):\n",
      "“attempt to set 'sep' ignored”\n"
     ]
    }
   ],
   "source": [
    "write.csv(cox_df,paste0(save_dir,\"SuppTable13_EFS.csv\"), sep=\",\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "double-massachusetts",
   "metadata": {},
   "source": [
    "<br>\n",
    "\n",
    "<br>\n",
    "\n",
    "# SuppFig 6A & SuppTable12"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "nominated-motorcycle",
   "metadata": {},
   "source": [
    "### Get transcriptional subtype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "dutch-running",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "ETP-like    Other \n",
       "     200      975 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "UPenn_dge$samples$Subtype <- as.vector(setNames(bulk_metadat_1$Reviewed.subtype,\n",
    "                                bulk_metadat_1$sample)[UPenn_dge$samples$sample_id])\n",
    "UPenn_dge$samples$transcriptional_ETP <- ifelse(UPenn_dge$samples$Subtype==\"ETP-like\",\"ETP-like\", \"Other\")\n",
    "\n",
    "table(UPenn_dge$samples$transcriptional_ETP)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "portable-election",
   "metadata": {},
   "source": [
    "### Split data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "burning-slovak",
   "metadata": {},
   "outputs": [],
   "source": [
    "cutquantile = 0.3333\n",
    "\n",
    "## -- By ZBTB16\n",
    "\n",
    "scvec_Z = UPenn_dge$logCPM[\"ZBTB16\",]\n",
    "scstrata_Z = rep(NA, length(scvec_Z))\n",
    "scstrata_Z[scvec_Z < quantile(scvec_Z, cutquantile)] = 'Low'\n",
    "scstrata_Z[scvec_Z > quantile(scvec_Z, 1 - cutquantile)] = 'High'\n",
    "\n",
    "## -- Rename\n",
    "scstrata_Z = gsub('High', ' ^ Hi', scstrata_Z)\n",
    "scstrata_Z = gsub('Low', ' ^ Lo', scstrata_Z)\n",
    "names(scstrata_Z) <- names(scvec_Z)\n",
    "\n",
    "UPenn_dge$samples$survival_group <- as.vector(scstrata_Z[UPenn_dge$samples$sample_id])\n",
    "UPenn_dge$samples$ZBTB16_exprs <-  as.vector(UPenn_dge$logCPM[\"ZBTB16\",])\n",
    "\n",
    "## -- By ETP  // exclude Near-ETP & Unknown\n",
    "\n",
    "scstrata_ETP = rep(NA, nrow(UPenn_dge$samples))\n",
    "scstrata_ETP[UPenn_dge$samples$transcriptional_ETP==\"ETP-like\"] = 'ETP-like'\n",
    "scstrata_ETP[UPenn_dge$samples$transcriptional_ETP==\"Other\"] = 'Other'\n",
    "names(scstrata_ETP) <- rownames(UPenn_dge$samples)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "communist-brass",
   "metadata": {},
   "source": [
    "### Run cox"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "objective-powell",
   "metadata": {},
   "outputs": [],
   "source": [
    "testCOX <- function(ZBTB16_vec_x, ETP_vec_x, status_str=\"OS.status\", event_str = \"OS\"){\n",
    "    ### Alternative:\n",
    "    # - status_str = \"EFS.status\"\n",
    "    # - event_str = \"EFS\"\n",
    "    \n",
    "    \n",
    "    ## -- Prep\n",
    "    cdata_x = UPenn_dge$samples[rownames(UPenn_dge$samples) %in% names(ETP_vec_x[!(is.na(ETP_vec_x))]),]\n",
    "\n",
    "\n",
    "\n",
    "    ## -- Check values\n",
    "    if (length(cdata_x[which((cdata_x[, event_str] < 0) == TRUE), \n",
    "                   \"short_name\"]) > 0 ){\n",
    "            print(\"Some patients have negative values for event\")\n",
    "    }\n",
    "    \n",
    "    # -- Create into survival data structure\n",
    "    cdata_x$s = grepl(\"1\", cdata_x[[status_str]], ignore.case = TRUE)\n",
    "    cdata_x$ETP = factor(as.vector(ETP_vec_x[rownames(cdata_x)]), levels=unique(as.vector(ETP_vec_x[rownames(cdata_x)])))\n",
    "\n",
    "    if (is.character(ZBTB16_vec_x[[1]])){\n",
    "        cdata_x$Z_expres <- factor(as.vector(ZBTB16_vec_x[rownames(cdata_x)]))\n",
    "        cdata_x$Z_expres <- relevel(cdata_x$Z_expres, \" ^ Lo\")\n",
    "    } else {\n",
    "        cdata_x$Z_expres = as.vector(ZBTB16_vec_x[rownames(cdata_x)])\n",
    "    }\n",
    "\n",
    "    cdata_x$event <- as.vector(cdata_x[[event_str]])\n",
    "    cdata_x = cdata_x[, c(\"event\", \"s\", \"ETP\",\"Z_expres\")]\n",
    "    cdata_x = cdata_x[order(cdata_x$ETP),]\n",
    "\n",
    "    #if (class(cdata_x$Z_expres)==\"numeric\"){\n",
    "    #    Z_str = \"Z_expres\"\n",
    "    #} else {\n",
    "    #    cdata_x$Z_expres <- as.factor(cdata_x$Z_expres)\n",
    "    #    cdata_x = cdata_x[rev(order(cdata_x$Z_expres)),]\n",
    "    #    Z_str = paste0(\"Z_expres\",unique(cdata_x$Z_expres[!(is.na(cdata_x$Z_expres))])[2])    \n",
    "    #}\n",
    "    \n",
    "    ## -- Set formula\n",
    "    f = as.formula(\"Surv(event, s) ~ ETP + Z_expres\")\n",
    "    \n",
    "    sfit = do.call(survival::coxph, list(formula = f, data = cdata_x))\n",
    "    \n",
    "    res.cox <- coxph(Surv(event, s) ~ ETP + Z_expres, data =  cdata_x,na.action=na.exclude)\n",
    "    \n",
    "   # print(Z_str)\n",
    "    print(summary(res.cox))\n",
    "    \n",
    "\n",
    "    p_x = summary(res.cox)$coefficients[,5][[2]]\n",
    "    p_y = summary(res.cox)$coefficients[,5][[1]]\n",
    "    coef_x = summary(res.cox)$conf.int[,1][[2]]\n",
    "    coef_ETP_x = summary(res.cox)$conf.int[,1][[\"ETPETP-like\"]]\n",
    "\n",
    "    lower_x  = summary(res.cox)$conf.int[,3][[2]]\n",
    "    upper_x = summary(res.cox)$conf.int[,4][[2]]\n",
    "    lower_ETP_x = summary(res.cox)$conf.int[,3][[\"ETPETP-like\"]]\n",
    "    upper_ETP_x = summary(res.cox)$conf.int[,4][[\"ETPETP-like\"]]\n",
    "\n",
    "    signif_x = 3\n",
    "    return(c(signif(p_x, signif_x), signif(p_y, signif_x), \n",
    "             signif(coef_x, signif_x), signif(coef_ETP_x, signif_x),\n",
    "             signif(lower_x, signif_x),signif(upper_x, signif_x),\n",
    "             signif(lower_ETP_x, signif_x), signif(upper_ETP_x, signif_x)))\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "egyptian-paste",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Call:\n",
      "coxph(formula = Surv(event, s) ~ ETP + Z_expres, data = cdata_x, \n",
      "    na.action = na.exclude)\n",
      "\n",
      "  n= 1175, number of events= 127 \n",
      "\n",
      "               coef exp(coef) se(coef)     z Pr(>|z|)   \n",
      "ETPETP-like 0.22412   1.25123  0.24213 0.926  0.35464   \n",
      "Z_expres    0.10729   1.11325  0.03711 2.891  0.00384 **\n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "            exp(coef) exp(-coef) lower .95 upper .95\n",
      "ETPETP-like     1.251     0.7992    0.7784     2.011\n",
      "Z_expres        1.113     0.8983    1.0352     1.197\n",
      "\n",
      "Concordance= 0.604  (se = 0.025 )\n",
      "Likelihood ratio test= 16.9  on 2 df,   p=2e-04\n",
      "Wald test            = 17.83  on 2 df,   p=1e-04\n",
      "Score (logrank) test = 18.37  on 2 df,   p=1e-04\n",
      "\n",
      "Call:\n",
      "coxph(formula = Surv(event, s) ~ ETP + Z_expres, data = cdata_x, \n",
      "    na.action = na.exclude)\n",
      "\n",
      "  n= 784, number of events= 92 \n",
      "   (391 observations deleted due to missingness)\n",
      "\n",
      "                coef exp(coef) se(coef)     z Pr(>|z|)   \n",
      "ETPETP-like   0.2690    1.3087   0.2438 1.103  0.26990   \n",
      "Z_expres ^ Hi 0.7247    2.0641   0.2500 2.899  0.00374 **\n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "              exp(coef) exp(-coef) lower .95 upper .95\n",
      "ETPETP-like       1.309     0.7641    0.8115     2.110\n",
      "Z_expres ^ Hi     2.064     0.4845    1.2646     3.369\n",
      "\n",
      "Concordance= 0.61  (se = 0.027 )\n",
      "Likelihood ratio test= 16.28  on 2 df,   p=3e-04\n",
      "Wald test            = 15.43  on 2 df,   p=4e-04\n",
      "Score (logrank) test = 16.41  on 2 df,   p=3e-04\n",
      "\n",
      "Call:\n",
      "coxph(formula = Surv(event, s) ~ ETP + Z_expres, data = cdata_x, \n",
      "    na.action = na.exclude)\n",
      "\n",
      "  n= 1175, number of events= 194 \n",
      "\n",
      "               coef exp(coef) se(coef)     z Pr(>|z|)   \n",
      "ETPETP-like 0.47785   1.61260  0.19274 2.479  0.01317 * \n",
      "Z_expres    0.09098   1.09525  0.03038 2.995  0.00275 **\n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "            exp(coef) exp(-coef) lower .95 upper .95\n",
      "ETPETP-like     1.613     0.6201     1.105     2.353\n",
      "Z_expres        1.095     0.9130     1.032     1.162\n",
      "\n",
      "Concordance= 0.611  (se = 0.021 )\n",
      "Likelihood ratio test= 31.77  on 2 df,   p=1e-07\n",
      "Wald test            = 34.86  on 2 df,   p=3e-08\n",
      "Score (logrank) test = 36.59  on 2 df,   p=1e-08\n",
      "\n",
      "Call:\n",
      "coxph(formula = Surv(event, s) ~ ETP + Z_expres, data = cdata_x, \n",
      "    na.action = na.exclude)\n",
      "\n",
      "  n= 784, number of events= 139 \n",
      "   (391 observations deleted due to missingness)\n",
      "\n",
      "                coef exp(coef) se(coef)     z Pr(>|z|)   \n",
      "ETPETP-like   0.6293    1.8762   0.2010 3.130  0.00175 **\n",
      "Z_expres ^ Hi 0.4654    1.5926   0.2054 2.266  0.02344 * \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "              exp(coef) exp(-coef) lower .95 upper .95\n",
      "ETPETP-like       1.876     0.5330     1.265     2.782\n",
      "Z_expres ^ Hi     1.593     0.6279     1.065     2.382\n",
      "\n",
      "Concordance= 0.625  (se = 0.023 )\n",
      "Likelihood ratio test= 28  on 2 df,   p=8e-07\n",
      "Wald test            = 29.08  on 2 df,   p=5e-07\n",
      "Score (logrank) test = 31.21  on 2 df,   p=2e-07\n",
      "\n"
     ]
    }
   ],
   "source": [
    "vec_1 = testCOX(scvec_Z, scstrata_ETP, status_str=\"OS.status\", event_str = \"OS\")\n",
    "#vec_2 = testCOX(scvec_Z, scstrata_ETP, status_str=\"OS.status\", event_str = \"OS\")\n",
    "vec_3 = testCOX(scstrata_Z, scstrata_ETP, status_str=\"OS.status\", event_str = \"OS\")\n",
    "#vec_4 = testCOX(scstrata_Z, scstrata_ETP, status_str=\"OS.status\", event_str = \"OS\")\n",
    "\n",
    "vec_5 = testCOX(scvec_Z, scstrata_ETP, status_str=\"EFS.status\", event_str = \"EFS\")\n",
    "#vec_6 = testCOX(scvec_Z, scstrata_ETP, status_str=\"EFS.status\", event_str = \"EFS\")\n",
    "vec_7 = testCOX(scstrata_Z, scstrata_ETP, status_str=\"EFS.status\", event_str = \"EFS\")\n",
    "#vec_8 = testCOX(scstrata_Z, scstrata_ETP, status_str=\"EFS.status\", event_str = \"EFS\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "broken-retail",
   "metadata": {},
   "source": [
    "### Save"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "accepted-mathematics",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 4 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>P-value</th><th scope=col>Coef. ZBTB16</th><th scope=col>Lower .95 ZBTB16</th><th scope=col>Higher .95 ZBTB16</th><th scope=col>P-value (ETP)</th><th scope=col>Coef. ETP</th><th scope=col>Lower .95 ETP</th><th scope=col>Higher .95 ETP</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>a_Z_cont_OS</th><td>0.00384</td><td>1.11</td><td>1.04</td><td>1.20</td><td>0.35500</td><td>1.25</td><td>0.778</td><td>2.01</td></tr>\n",
       "\t<tr><th scope=row>b_Z_tert_OS</th><td>0.00374</td><td>2.06</td><td>1.26</td><td>3.37</td><td>0.27000</td><td>1.31</td><td>0.811</td><td>2.11</td></tr>\n",
       "\t<tr><th scope=row>c_Z_cont_EFS</th><td>0.00275</td><td>1.10</td><td>1.03</td><td>1.16</td><td>0.01320</td><td>1.61</td><td>1.110</td><td>2.35</td></tr>\n",
       "\t<tr><th scope=row>d_Z_tert_EFS</th><td>0.02340</td><td>1.59</td><td>1.06</td><td>2.38</td><td>0.00175</td><td>1.88</td><td>1.270</td><td>2.78</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 4 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & P-value & Coef. ZBTB16 & Lower .95 ZBTB16 & Higher .95 ZBTB16 & P-value (ETP) & Coef. ETP & Lower .95 ETP & Higher .95 ETP\\\\\n",
       "\\hline\n",
       "\ta\\_Z\\_cont\\_OS & 0.00384 & 1.11 & 1.04 & 1.20 & 0.35500 & 1.25 & 0.778 & 2.01\\\\\n",
       "\tb\\_Z\\_tert\\_OS & 0.00374 & 2.06 & 1.26 & 3.37 & 0.27000 & 1.31 & 0.811 & 2.11\\\\\n",
       "\tc\\_Z\\_cont\\_EFS & 0.00275 & 1.10 & 1.03 & 1.16 & 0.01320 & 1.61 & 1.110 & 2.35\\\\\n",
       "\td\\_Z\\_tert\\_EFS & 0.02340 & 1.59 & 1.06 & 2.38 & 0.00175 & 1.88 & 1.270 & 2.78\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 4 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | P-value | Coef. ZBTB16 | Lower .95 ZBTB16 | Higher .95 ZBTB16 | P-value (ETP) | Coef. ETP | Lower .95 ETP | Higher .95 ETP |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| a_Z_cont_OS | 0.00384 | 1.11 | 1.04 | 1.20 | 0.35500 | 1.25 | 0.778 | 2.01 |\n",
       "| b_Z_tert_OS | 0.00374 | 2.06 | 1.26 | 3.37 | 0.27000 | 1.31 | 0.811 | 2.11 |\n",
       "| c_Z_cont_EFS | 0.00275 | 1.10 | 1.03 | 1.16 | 0.01320 | 1.61 | 1.110 | 2.35 |\n",
       "| d_Z_tert_EFS | 0.02340 | 1.59 | 1.06 | 2.38 | 0.00175 | 1.88 | 1.270 | 2.78 |\n",
       "\n"
      ],
      "text/plain": [
       "             P-value Coef. ZBTB16 Lower .95 ZBTB16 Higher .95 ZBTB16\n",
       "a_Z_cont_OS  0.00384 1.11         1.04             1.20             \n",
       "b_Z_tert_OS  0.00374 2.06         1.26             3.37             \n",
       "c_Z_cont_EFS 0.00275 1.10         1.03             1.16             \n",
       "d_Z_tert_EFS 0.02340 1.59         1.06             2.38             \n",
       "             P-value (ETP) Coef. ETP Lower .95 ETP Higher .95 ETP\n",
       "a_Z_cont_OS  0.35500       1.25      0.778         2.01          \n",
       "b_Z_tert_OS  0.27000       1.31      0.811         2.11          \n",
       "c_Z_cont_EFS 0.01320       1.61      1.110         2.35          \n",
       "d_Z_tert_EFS 0.00175       1.88      1.270         2.78          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cox_df = data.frame(a_Z_cont_OS = vec_1,\n",
    "          # b_Z_cont_OS_near = vec_2,\n",
    "           b_Z_tert_OS = vec_3,\n",
    "         # d_Z_tert_near = vec_4,\n",
    "           c_Z_cont_EFS = vec_5,\n",
    "          # f_Z_EFS_cont_near = vec_6,\n",
    "           d_Z_tert_EFS = vec_7,\n",
    "         #  h_Z_tert_EFS_near = vec_8,\n",
    "                    row.names = c(\"P-value\", \"P-value (ETP)\", \"Coef. ZBTB16\", \"Coef. ETP\",\n",
    "                                            \"Lower .95 ZBTB16\", \"Higher .95 ZBTB16\",\n",
    "                                             \"Lower .95 ETP\", \"Higher .95 ETP\"))\n",
    "cox_df <- t(cox_df)\n",
    "\n",
    "column_ordering <- c(\"P-value\", \"Coef. ZBTB16\", \"Lower .95 ZBTB16\", \"Higher .95 ZBTB16\",\"P-value (ETP)\", \"Coef. ETP\", \"Lower .95 ETP\", \"Higher .95 ETP\")\n",
    "\n",
    "cox_df[,column_ordering]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "educated-initial",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in write.csv(cox_df, paste0(save_dir, \"SuppTable12_and_SuppFig6A.csv\"), :\n",
      "“attempt to set 'sep' ignored”\n"
     ]
    }
   ],
   "source": [
    "write.csv(cox_df,paste0(save_dir,\"SuppTable12_and_SuppFig6A.csv\"), sep=\",\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "brief-representation",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Single Cell R (4.3)",
   "language": "R",
   "name": "scr_4.3"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
